tuna/miopen/worker/fin_class.py

```diff
@@ -348,52 +348,57 @@ def __rm_old_app(self, session: DbSession, cfg_rows: list) -> None:
     session.commit()
     self.logger.info("Finished applic zeroing")
 
+  def query_cfgs(self, label=None):
+    """query all configs from table, optionally limit by label"""
+    with DbSession() as session:
+      query = session.query(self.dbt.config_table)\
+                        .filter(self.dbt.config_table.valid == 1)
+
+      if label:
+        query = query.filter(self.dbt.config_table.id == self.dbt.config_tags_table.config)\
+            .filter(self.dbt.config_tags_table.tag == label)
+
+      #order by id for splitting configs into blocks
+      query = query.order_by(self.dbt.config_table.id)
+      return query
+
   def __set_all_configs(self, idx: int = 0, num_blk: int = 1) -> bool:
     """Gathering all configs from Tuna DB to set up fin input file"""
     if idx == 0:
-      with DbSession() as session:
-        query = session.query(self.dbt.config_table)\
-                          .filter(self.dbt.config_table.valid == 1)
-
-        if self.label:
-          query = query.filter(self.dbt.config_table.id == self.dbt.config_tags_table.config)\
-              .filter(self.dbt.config_tags_table.tag == self.label)
+      query = self.query_cfgs(self.label)
+      rows = query.all()
 
-        #order by id for splitting configs into blocks
-        query = query.order_by(self.dbt.config_table.id)
-        rows = query.all()
-
-        len_rows = len(rows)
-        master_cfg_list = []
-        for row in rows:
-          r_dict = compose_config_obj(row, self.config_type)
-          if self.config_type == ConfigType.batch_norm:
-            r_dict['direction'] = row.get_direction()
-          master_cfg_list.append(r_dict)
+      len_rows = len(rows)
+      master_cfg_list = []
+      for row in rows:
+        r_dict = compose_config_obj(row, self.config_type)
+        if self.config_type == ConfigType.batch_norm:
+          r_dict['direction'] = row.get_direction()
```

Comment:
can this be moved inside compose_config_obj(), otherwise it will have to be accounted for separately which will definitely lead to oversights

Reply:
out of scope